privval/grpc/server.go

```diff
@@ -0,0 +1,86 @@
+package grpc
+
+import (
+	context "context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/tendermint/tendermint/crypto"
+	cryptoenc "github.com/tendermint/tendermint/crypto/encoding"
+	"github.com/tendermint/tendermint/libs/log"
+	privvalproto "github.com/tendermint/tendermint/proto/tendermint/privval"
+	"github.com/tendermint/tendermint/types"
+)
+
+type SignerServer struct {
+	logger log.Logger
+
+	chainID string
+	privVal types.PrivValidator
+}
+
+func NewSignerServer(chainID string,
+	privVal types.PrivValidator, log log.Logger) *SignerServer {
+
+	return &SignerServer{
+		logger:  log,
+		chainID: chainID,
+		privVal: privVal,
+	}
+}
+
+var _ privvalproto.PrivValidatorAPIServer = (*SignerServer)(nil)
+
+// PubKey receives a request for the pubkey
+// returns the pubkey on success and error on failure
+func (ss *SignerServer) GetPubKey(ctx context.Context, req *privvalproto.PubKeyRequest) (
+	*privvalproto.PubKeyResponse, error) {
+	var pubKey crypto.PubKey
+
+	pubKey, err := ss.privVal.GetPubKey()
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "error getting pubkey: %v", err)
+	}
+
+	pk, err := cryptoenc.PubKeyToProto(pubKey)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "error transitioning pubkey to proto: %v", err)
+	}
+
+	ss.logger.Info("SignerServer: GetPubKey Success")
+
+	return &privvalproto.PubKeyResponse{PubKey: pk}, nil
+}
+
+// SignVote receives a vote sign requests, attempts to sign it
+// returns SignedVoteResponse on success and error on failure
+func (ss *SignerServer) SignVote(ctx context.Context, req *privvalproto.SignVoteRequest) (
+	*privvalproto.SignedVoteResponse, error) {
+	vote := req.Vote
+
+	err := ss.privVal.SignVote(req.ChainId, vote)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "error signing vote: %v", err)
+	}
+
+	ss.logger.Info("SignerServer: SignVote Success")
```

Comment:
do we want to print some details here? like vote's hash or something?

Reply:
Not sure that brings any value to someone using this. Personally I'd consider it noise to have hashes of vote, proposal and various other things in the Info log.

Reply:
u mean operators don't need to see what they're signing? okay

Reply:
Could you explain why someone would want to use it? I can't see a reason, but could be missing something. For me I wouldn't know what do with the outputted hash. 

I can add a debug with hashes and other information.